### PR TITLE
Tickets/sitcom 1345

### PR DESCRIPTION
**Pull request by Kelks19**
This is a basic FFT analysis that I had hoped to reveal dominant frequencies in the TEA vibrations. However, the analysis did not give us conclusive results.

### CODE REVIEW

File: notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,312 @@
+{
```
Comment (boutigny) [inline notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    from scipy import fftpack
</pre><p>fftpack is a legacy package and is not maintained anymore. I suggest to use the fft library from scipy or from numpy</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/138/discussion/#comment-1634816623'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (boutigny) [inline notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    n= str('2024-03-10_13.13.06') # event instances mentioned above number of the audio file.  
</pre><p>Be careful, the audio files are stereo recording, you need to extract both channels, you can use this online tool to do that: <a href="https://audioalter.com/downmixer" target="_blank">https://audioalter.com/downmixer</a></p><p>By keeping both audio channels, I guess that you are doubling the samples</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/138/discussion/#comment-1634816629'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

Comment (boutigny) [inline notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    raw = open_wave('Audio_'+n+'.wav','rb')
</pre><p>I suggest to format the string using "f" and "{   }"</p><p>raw = open_wave(f'Audio_{n}.wav','rb')</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/138/discussion/#comment-1634816641'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

Comment (boutigny) [inline notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    sample_freq = fftpack.fftfreq(len(signal),time)
</pre><p>I think that you need to multiply sample_freq by the sample rate in order to get the frequency in Hertz.</p><p>If you do that you will see that the base frequency is around 3 kHz which correspond to the noise that can be heard in the audio file.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/138/discussion/#comment-1634816647'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='11'/>

**Review by boutigny — COMMENTED** [review]
I suggest to investigate the possibility to compute the PSD instead of the FFT.
I also think that it would be useful to make spectrogram (frequency versus time). To do so, you can do the following:

from scipy.signal import stft
f, t, Zxx = stft(signal, f_rate, "hamming", 128)
plt.pcolormesh(t, f, np.abs(Zxx), vmin=0, vmax=10, shading="gouraud")

parameters in stft, may need to be adjusted.

File: notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb (hunk 1 above)
Comment (Kelks19) [reply notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: Thanks! I think this is what I missed while cross checking.

Comment (Kelks19) [reply notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: I was not aware of this (never worked recorded audio files before )- Thanks for the tip!

Comment (Kelks19) [reply notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: ok !

**Comment by Kelks19** [comment]
Let me do the following - 
1. update the original FFT analysis incorporating your comments 
2. Include PSD analysis so that I can compare the methods. 

We can then decide to keep one in the final code.

File: notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb (hunk 1 above)
Comment (Kelks19) [reply notebooks/tel_and_site/subsys_req_ver/tma/SITCOM-1345_TEA_Vibration_audio_freq_analysis.ipynb:1]: Updated !

**Comment by Kelks19** [comment]
I have pushed all the changes according to the feedback above. 
The spectrogram is a good diagnostic to identify temporal vibrations - although we may not have any clear indications yet (there a few temporal detections for certain frequencies for one of the audio clips. What do you think ? )

**Review by boutigny — APPROVED** [review]
I am accepting the PR but as disscussed on Slack, I suggest to change the frequency scale of the spectrogram from [0, 100] to [0, 2000]

**Pull request merged**